### PR TITLE
Disk space alerts

### PR DESCRIPTION
**Pull request by MancunianSam**
Modules changes to support https://github.com/nationalarchives/tdr-jenkins/pull/209

### CODE REVIEW

File: cloudwatch_alarms/variables.tf [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+variable "period" {
+  default     = 60
+  description = "The period in seconds over which the specified statistic is applied."
+}
+variable "datapoints_to_alarm" {
+  default     = 1
+  description = "How many datapoints over the threshold to trigger the alarm"
+}
+variable "threshold" {
+  description = "The threshold for the metric above which is will alarm"
```
Comment (TomJKing) [inline cloudwatch_alarms/variables.tf:19]: Maybe reword description: "The threshold for the metric above which **an alarm is triggered**"

File: cloudwatch_alarms/variables.tf [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  description = "Can be SampleCount, Average, Sum, Minimum, Maximum"
+}
+
+variable "comparison_operator" {
+  default     = "GreaterThanThreshold"
+  description = "Can be GreaterThanOrEqualToThreshold, GreaterThanThreshold, LessThanThreshold, LessThanOrEqualToThreshold"
+}
+variable "treat_missing_data" {
+  default     = "breaching"
+  description = "Sets how this alarm is to handle missing data points. Can be missing, ignore, breaching and notBreaching. Default is breaching so we'll get an alert if cloudwatch agent fails"
```
Comment (TomJKing) [inline cloudwatch_alarms/variables.tf:53]: Maybe reword the description slightly: "... Default is 'breaching' so there will be an alert if the Cloudwatch agent fails"

File: ec2/templates/user_data_jenkins_docker.sh.tpl [hunk 3]
```diff
@@ -1,3 +1,5 @@
 #!/usr/bin/env bash
 echo ECS_CLUSTER=${jenkins_cluster_name} > /etc/ecs/ecs.config
-chown 1000:1000 /var/run/docker*
\ No newline at end of file
+chown 1000:1000 /var/run/docker*
+yum install -y amazon-cloudwatch-agent
+amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -s -c ssm:${agent_policy_parameter_name}
```
Comment (TomJKing) [inline ec2/templates/user_data_jenkins_docker.sh.tpl:5]: Missing line break

File: iam_policy/templates/jenkins_cloudwatch_agent_integration.json.tpl [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "cloudwatch:PutMetricData",
+        "ec2:DescribeVolumes",
+        "ec2:DescribeTags",
+        "logs:PutLogEvents",
```
Comment (TomJKing) [inline iam_policy/templates/jenkins_cloudwatch_agent_integration.json.tpl:10]: Maybe order "Action" alphabetically

File: iam_policy/templates/jenkins_cloudwatch_agent_integration.json.tpl [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (TomJKing) [inline iam_policy/templates/jenkins_cloudwatch_agent_integration.json.tpl:1]: Should this policy be renamed: `jenkins_cloudwatch_agent.json.tpl`. Looks like it will be used both for the integration and prod Jenkins instances?

**Review by TomJKing — COMMENTED** [review]
Looks good. Some minor comments

File: cloudwatch_alarms/variables.tf (hunk 1 above)
Comment (MancunianSam) [reply cloudwatch_alarms/variables.tf:19]: What I've written isn't even proper English. I've changed it to your wording.

File: cloudwatch_alarms/variables.tf (hunk 2 above)
Comment (MancunianSam) [reply cloudwatch_alarms/variables.tf:53]: I've changed this one as well.

File: ec2/templates/user_data_jenkins_docker.sh.tpl (hunk 3 above)
Comment (MancunianSam) [reply ec2/templates/user_data_jenkins_docker.sh.tpl:5]: I found this comment https://github.com/nationalarchives/tdr-terraform-modules/pull/106/files#r636863512 where I said I was going to do it around May so I think it's probably time now. I've added on in.

File: iam_policy/templates/jenkins_cloudwatch_agent_integration.json.tpl (hunk 5 above)
Comment (MancunianSam) [reply iam_policy/templates/jenkins_cloudwatch_agent_integration.json.tpl:1]: Good point, I've renamed it.

**Review by TomJKing — APPROVED** [review]
